src/ArmTemplates/Extractor/Models/ExtractorParameters.cs

```diff
@@ -1,56 +1,133 @@
-﻿namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.FileHandlers;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models
 {
-    public class ExtractorParameters
+    /// <summary>
+    /// Parameters used for extractor flow. 
+    /// They are not changed in the process of extraction.
+    /// </summary>
+    public record ExtractorParameters
     {
-        public string sourceApimName { get; private set; }
-        public string destinationApimName { get; private set; }
-        public string resourceGroup { get; private set; }
-        public string fileFolder { get; private set; }
-        public string linkedTemplatesBaseUrl { get; private set; }
-        public string linkedTemplatesSasToken { get; private set; }
-        public string linkedTemplatesUrlQueryString { get; private set; }
-        public string policyXMLBaseUrl { get; private set; }
-        public string policyXMLSasToken { get; private set; }
-        public string apiVersionSetName { get; private set; }
-        public bool includeAllRevisions { get; private set; }
-        public serviceUrlProperty[] serviceUrlParameters { get; set; }
-        public bool paramServiceUrl { get; private set; }
-        public bool paramNamedValue { get; private set; }
-        public bool paramApiLoggerId { get; private set; }
-        public bool paramLogResourceId { get; private set; }
-        public bool notIncludeNamedValue { get; private set; }
-        public bool paramNamedValuesKeyVaultSecrets { get; private set; }
-
-        public int operationBatchSize { get; private set; }
-
-        public bool paramBackend { get; set; }
-
-        public ExtractorParameters(ExtractorConfig exc, string dirName)
+        public string SourceApimName { get; private set; }
+
+        public string DestinationApimName { get; private set; }
+
+        public string ResourceGroup { get; private set; }
+
+        /// <summary>
+        /// Naming of all templates from the base folder requested by end-user in extractor-configuration parameters
+        /// </summary>
+        public FileNames FileNames { get; private set; }
+
+        /// <summary>
+        /// Not changeable root directory, where templates will be generated.
+        /// </summary>
+        public string FilesGenerationRootDirectory { get; private set; }
+
+        /// <summary>
+        /// Name of a single API that user wants to extract
+        /// </summary>
+        public string SingleApiName { get; set; }
+
+        /// <summary>
+        /// Names of APIs that user wants to extract
+        /// </summary>
+        public List<string> MultipleApiNames { get; set; }
+
+        /// <summary>
+        /// Create split api templates for every API in the source APIM instance
```

Comment:
Do you mean separate api templates?

Reply:
As i understand, if you are not setting `splitapi`, then all apis in APIM instance are loaded in one file. Ye, I think it is better to write `separate api templates`, thanks